src/mconnect/mpris.vala

```diff
@@ -0,0 +1,399 @@
+        }
+        if (prop.album_art_url != null) {
+            builder.set_member_name ("albumArtUrl");
+            builder.add_string_value (prop.album_art_url);
+        }
+        if (prop.length != null) {
+            builder.set_member_name ("length");
+            builder.add_int_value (prop.length);
+        }
+        if (prop.is_playing != null) {
+            builder.set_member_name ("isPlaying");
+            builder.add_boolean_value (prop.is_playing);
+        }
+        if (prop.can_pause != null) {
+            builder.set_member_name ("canPause");
+            builder.add_boolean_value (prop.can_pause);
+        }
+        if (prop.can_play != null) {
+            builder.set_member_name ("canPlay");
+            builder.add_boolean_value (prop.can_play);
+        }
+        if (prop.can_go_next != null) {
+            builder.set_member_name ("canGoNext");
+            builder.add_boolean_value (prop.can_pause);
+        }
+        if (prop.can_go_previous != null) {
+            builder.set_member_name ("canGoPrevious");
+            builder.add_boolean_value (prop.can_pause);
+        }
+        if (prop.can_seek != null) {
+            builder.set_member_name ("canSeek");
+            builder.add_boolean_value (prop.can_pause);
+        }
+        builder.set_member_name ("pos");
+        builder.add_int_value (prop.position);
+        builder.end_object ();
+        return new Packet (MPRIS_PKT,
+                           builder.get_root ().get_object ());
+    }
+
+    private void add_to_player_list (string bus_name) {
+        debug ("mpris player found: %s", bus_name);
+        try {
+            MprisProxy mpris = Bus.get_proxy_sync (BusType.SESSION,
+                                                   bus_name,
+                                                   "/org/mpris/MediaPlayer2");
+            player_list.insert (mpris.identity, bus_name);
+        } catch (IOError e) {
+            warning ("failed to connect to mpris player: %s", e.message);
+        }
+    }
+
+    private void get_player_list () {
+        if (_dbus_watcher == null) {
+            try {
+                _dbus_watcher = Bus.get_proxy_sync (BusType.SESSION, "org.freedesktop.DBus", "/org/freedesktop/DBus");
+
+                _dbus_watcher.name_owner_changed.connect ((name, old_owner, new_owner) => {
+                    if (!name.has_prefix ("org.mpris.MediaPlayer2.")) {
+                        return;
+                    }
+                    if (new_owner == "") {
+                        debug ("mpris player disconnected: %s", name);
+                        string key = "";
+                        player_list.find ((k, v) => {
+                            if (v == name) {
+                                key = k;
+                            }
+                            return v == k;
+                        });
+                        player_list.remove (key);
+                        _properties_watchers.remove (key);
```

Comment:
If I understood correctly, the mappings are:
```
player_list: player_id (eg. spotify)  -> DBus name org.mpris.MediaPlayer2.spotify
_properties_watchers: DBus name -> properties proxy
```
While here you are using player_id (eg. spotify) to remove the watcher.